app/client/src/utils/hooks.js

```diff
@@ -32,6 +33,59 @@ function closePopup({ mapView, setHighlightedGraphic, setSelectedGraphic }) {
   if (mapView) mapView.popup.close();
 }
 
+// Gets all features in the layer that match the provided organizationid and
+// assessmentunitidentifier. Any features found are added to the provided
+// features array.
+function getMatchingFeatures(
+  features,
+  layerData,
+  organizationid,
+  assessmentunitidentifier,
+) {
+  layerData.features.forEach((feature) => {
+    if (
+      feature.attributes.organizationid === organizationid &&
+      feature.attributes.assessmentunitidentifier === assessmentunitidentifier
+    ) {
+      features.push(feature);
+    }
+  });
+}
+
+// Highlights the feature either by adding a graphic to the view.graphics
+// collection or by using the layerView.highlight function. The view.graphics
+// method is used if the feature has an "originalGeometry" attribute, which
+// indicates that this feature has been clipped and the highlighting should
+// use the full original feature.
+function highlightFeature({
+  mapView,
+  features,
+  highlightOptions,
+  handles,
+  group,
+  layer = null,
+  callback = null,
+}) {
+  features.forEach((feature) => {
+    if (feature.originalGeometry) {
+      mapView.graphics.add({
+        ...feature,
+        geometry: feature.originalGeometry,
+        symbol: getHighlightSymbol(feature.originalGeometry, highlightOptions),
```

Comment:
Since the "color" property of `highlightOptions` is all that's used in the `getHighlightSymbol()` function, I'd recommend only passing that property along here (good practice in general to only pass data that you need).

```suggestion
        symbol: getHighlightSymbol(feature.originalGeometry, highlightOptions.color),
```